### PR TITLE
Expose vector & domain types in discretization

### PR DESCRIPTION
**Pull request by jbigot**

### CODE REVIEW

File: include/ddc/uniform_discretization.hpp [hunk 1]
```diff
@@ -51,6 +55,8 @@ class UniformDiscretization
      * @param a the coordinate of a first real point (will have mesh coordinate 0)
      * @param b the coordinate of the second real point (will have mesh coordinate `n-1`)
      * @param n the number of points to map the segment \f$[a, b]\f$ including a & b
+     * 
+     * @deprecated use the version accepting a vector for n instead
      */
     constexpr UniformDiscretization(rcoord_type a, rcoord_type b, std::size_t n)
```
Comment (tpadioleau) [inline include/ddc/uniform_discretization.hpp:61]: Should we mark it `[[deprecated]]` ?

**Review by tpadioleau — DISMISSED** [review]

File: include/ddc/uniform_discretization.hpp (hunk 1 above)
Comment (jbigot) [reply include/ddc/uniform_discretization.hpp:61]: As long as our only user is Voice, I even think we should remove it ASAP

File: include/ddc/discrete_domain.hpp [hunk 2]
```diff
@@ -22,6 +22,8 @@ class DiscreteDomain
     template <class...>
     friend class DiscreteDomain;
 
+    friend std::tuple_element_t<0, std::tuple<DDims...>>;
+
```
Comment (jbigot) [inline include/ddc/discrete_domain.hpp:26]:
```suggestion
```

File: include/ddc/discrete_domain.hpp [hunk 3]
```diff
@@ -59,6 +61,8 @@ class DiscreteDomain
 
     /** Construct a DiscreteDomain starting from (0, ..., 0) with size points.
      * @param size the number of points in each dimension
+     * 
+     * @deprecated use the version with explicit lower bound instead
      */
     constexpr DiscreteDomain(mlength_type const& size)
```
[the remark this answers is not in the document]
Comment (jbigot) [reply include/ddc/discrete_domain.hpp:65]:
```suggestion
    [[deprecated]] constexpr DiscreteDomain(mlength_type const& size)
```

**Pull request merged**